docs/tutorial/source-plugin-tutorial.md

```diff
@@ -0,0 +1,977 @@
+---
+title: "Creating a Source Plugin"
+tableOfContentsDepth: 2
+---
+
+In this tutorial, you'll create your own source plugin that will gather data from an API. The plugin will source data, optimize remote images, and create foreign key relationships between data sourced by your plugin.
+
+## What is a source plugin?
+
+Source plugins "source" data from remote or local locations into what Gatsby calls [nodes](/docs/node-interface/). This tutorial uses a demo API so that you can see how the data works on both the frontend and backend, but the same principles apply if you would like to source data from another API.
+
+For more background on source plugins, check out [Gatsby's source plugin documentation](/docs/creating-a-source-plugin/).
+
+## Why create a source plugin?
+
+Source plugins convert data from any source into a format that Gatsby can process. Your Gatsby site can use several source plugins to combine data in interesting ways.
+
+There may not be [an existing plugin](/plugins/?=gatsby-source) for your data source, so you can create your own.
+
+_**NOTE:** if your data is local i.e. on your file system and part of your site's repo, then you generally don't want to create a new source plugin. Instead you want to use [gatsby-source-filesystem](/packages/gatsby-source-filesystem/) which handles reading and watching files for you. You can then use [transformer plugins](/plugins/?=gatsby-transformer) like [gatsby-transformer-yaml](/packages/gatsby-transformer-yaml/) to make queryable data from files._
+
+## How to create a source plugin
+
+### Overview
+
+The plugin in this tutorial will source blog posts and authors from the demo API, link the posts and authors, and take image URLs from the posts and optimize them automatically. You'll be able to configure your plugin in your site's `gatsby-config.js` file and write GraphQL queries to access your plugin's data.
+
+This tutorial builds off of an existing Gatsby site and some data. If you want to follow along with this tutorial, you can find the codebase inside [the examples folder of the Gatsby repository](https://github.com/gatsbyjs/gatsby/tree/master/examples/creating-source-plugins). Once you clone this code, make sure to delete the `source-plugin` and `example-site` folders. Otherwise, the tutorial steps will already be completed.
```

Comment:
"Once you clone this code"

i miss a hint or a command how to clone a subdirectory. a new user did not have a clue how to do this. maybe add a git sparse checkout command for this?

Reply:
i added a new issue for this:

- #24547 `Improve tutorial "create a source plugin"`